include/frozen/bits/pmh.h

```diff
@@ -103,6 +103,11 @@ pmh_buckets<M> constexpr make_pmh_buckets(const carray<Item, N> & items,
     bool rejected = false;
     for (std::size_t i = 0; i < items.size(); ++i) {
       auto & bucket = result.buckets[hash(key(items[i]), static_cast<std::size_t>(result.seed)) % M];
+      for (const auto item_index : bucket) {
+        if (key(items[item_index]) == key(items[i])) {
```

Comment:
- This should use the `KeyEqual` parameter.
- This would probably be better if `key(items[i])` were saved in a temporary variable.
- move this loop to an helper function?
- Move the check under `ifndef NDEBUG` ?
- Instead of calling (`exit`), what about
```
+      extern void check(const char[]);
+      check("Duplicate keys present, check your input data");
```

Reply:
I agree with all of these except maybe the `ifndef NDEBUG` comment. For my use case, I'm writing a library where the keys will end up being supplied by the user. I think it would be better for their experience if they always saw a helpful message, even in debug mode. Perhaps we could add a macro that would force the check to happen, but by default it's only in debug mode? So something like `#if !defined NDEBUG || defined FROZEN_LETITGO_ENABLE_DUPLICATE_KEY_ASSERTIONS`?

Reply:
Also, if adding another preprocessor directive, and also since hopefully the old error message should go away in debug mode, should I update the README to mention the new macro?

Reply:
Just some running commentary: Extracting that check to a separate function might be more trouble than it's worth, due to the inputs it ends up needing to be heavily templated and type annotated. I'll try to get it working just to show what it would look like.

I'll need to pass KeyEqual down the stack quite a bit - though this is for correctness so that's fine.

Reply:
I've incorporated everything except the NDEBUG comment, once we reach a consensus I'll add that in.